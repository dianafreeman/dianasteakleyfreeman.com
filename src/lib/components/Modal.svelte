<script>
  import { trapFocus } from "$lib/actions";
  import Button from "./Buttons/Button.svelte";

  /** @type { string }*/
  let clazz;
  export { clazz as class };
  /** @type { string }*/
  export let title;
  /** @type { string }*/
  export let description;
  /** @type { string }*/
  export let buttonText;
  /** @type { string }*/
  export let buttonClasses;

  let isOpen = false;

  function toggleOpen() {
    isOpen = !isOpen;
  }

  const describedBy = description ? { "aria-describedby": "modal_desc" } : {};
  const ariaProps = {
    "aria-labelledby": "modal_label",
    "aria-modal": "true",
    ...describedBy
  };
</script>

<Button on:click={toggleOpen} class="w-fit p-5 {buttonClasses}"
  >{buttonText}</Button>
{#if isOpen}
  <div
    use:trapFocus={{ onEscPressed: toggleOpen }}
    id="modal"
    role="dialog"
    class="border border-gray {clazz}"
    class:hidden={!isOpen}
    {...ariaProps}>
    <div class="flex w-full justify-end">
      <button on:click={toggleOpen}>close</button>
    </div>
    <h2 id="modal_label" class="dialog_label sr-only">{title}</h2>

    {#if description}
      <div id="modal_desc">{description}</div>
    {/if}
    <slot />
  </div>
{/if}
