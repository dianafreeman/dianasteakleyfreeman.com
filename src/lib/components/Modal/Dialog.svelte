<script>
  import { trapFocus } from "$lib/actions";
  import { createEventDispatcher, onMount } from "svelte";

  let clazz;
  export { clazz as class };
  let closeButton;

  export let isOpen;
  export let title;
  export let ariaProps;

  const dispatch = createEventDispatcher();

  function onCloseButtonClick() {
    dispatch("closeClicked", { isOpen });
  }
  function onEscPressed(e) {
    dispatch("escPressed", { ...e });
  }

  onMount(() => {
    closeButton.focus();
  });
</script>

<div
  use:trapFocus={{ onEscPressed }}
  id="modal"
  role="dialog"
  class="bg-semi-transparent fixed top-0 left-0 z-[99] h-screen w-screen overflow-y-scroll p-16 shadow-lg {clazz} "
  class:hidden={!isOpen}
  {...ariaProps}>
  <div class="m-auto w-full bg-black pb-2">
    <div class="flex w-full justify-between rounded-t-md bg-gray">
      {#if title}
        <h2 id="modal_label" class="m-5 text-xl font-bold lowercase">
          {title}.
        </h2>
      {/if}
      <button
        bind:this={closeButton}
        class="py-2 px-4"
        on:click={onCloseButtonClick}>X</button>
    </div>
    <slot />
  </div>
</div>
