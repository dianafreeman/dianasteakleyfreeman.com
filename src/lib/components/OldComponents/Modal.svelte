<script>
  import { focusTrap } from "$lib/actions/focusTrap";
  import { settingsMenuIsOpen } from "$stores/LayoutStore";
  import Button from "../Buttons/Button.svelte";
  import { afterUpdate, createEventDispatcher, onMount } from "svelte";

  export let isOpen = false;
  export let id;
  export let title = "";
  export let description = "";
  export let onCloseClick;
  export let className = "";

  const dispatch = createEventDispatcher();

  const describedBy = description ? { "aria-describedby": "modal-desc" } : {};
  const ariaProps = {
    "aria-labelledby": "modal-title",
    "aria-modal": "true",
    ...describedBy
  };

  function handleClose() {
    isOpen = false;
    if (onCloseClick) onCloseClick();
    dispatch("closeClicked");
  }

  function handleEscape(event) {
    if (event.key === "Escape") {
      handleClose();
    }
  }

settingsMenuIsOpen.subscribe( bool => {
  console.log("running")
  if (bool == true) {
    const closeButton = document.querySelector(`#${id}-close`)
    console.log(closeButton)
    closeButton.focus()
  }
})
</script>

<div
  {id}
  role="dialog"
  class="bg-semi-transparent fixed top-0 left-0 z-[99] h-screen w-screen overflow-y-scroll p-16 shadow-lg {className}"
  aria-hidden={!isOpen}
  class:hidden={!isOpen}
  {...ariaProps}
  on:keydown={handleEscape}>
  <div class="m-auto w-full bg-black pb-2 rounded-md">
    <div class="flex w-full justify-between rounded-t-md bg-gray">
      {#if title}
        <h2 id="modal-title" class="m-5 text-xl font-bold lowercase">
          {title}
        </h2>
      {/if}
      <Button
        id="{id}-close"
        iconProps={{ type: "close" }}
        label="close"
        class="inline w-fit p-3 text-right"
        on:click={handleClose}>
        Close
      </Button>
    </div>
    <div class="p-4"
    use:focusTrap={isOpen}

    >
      <slot />
      {#if description}
        <div id="modal-desc" class="mt-2 text-sm text-gray-300">
          {description}
        </div>
      {/if}
    </div>
  </div>
</div>
