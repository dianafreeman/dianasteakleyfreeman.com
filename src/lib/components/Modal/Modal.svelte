<script>
  import Button from "../Buttons/Button.svelte";
  import Dialog from "./Dialog.svelte";

  /** @type { string }*/
  let clazz;
  export { clazz as class };
  /** @type { string }*/
  export let title;
  /** @type { string }*/
  export let description;
  /** @type { string }*/
  export let buttonText;
  /** @type { string }*/
  export let buttonClasses;
  /** @type { string }*/
  export let buttonIconClasses;
  /** @type { boolean }*/
  export let useCustomToggle;

  let isOpen = false;

  function toggleOpen() {
    console.log("toggling");
    isOpen = !isOpen;
  }

  const describedBy = description ? { "aria-describedby": "modal_desc" } : {};
  const ariaProps = {
    "aria-labelledby": "modal_label",
    "aria-modal": "true",
    ...describedBy
  };
</script>

{#if useCustomToggle}
  <button on:click={toggleOpen}>
    <slot name="toggle" />
  </button>
{:else}
  <Button
    on:click={toggleOpen}
    iconClasses={buttonIconClasses}
    class="w-fit p-5 {buttonClasses}">{buttonText}</Button>
{/if}
<Dialog
  {ariaProps}
  {isOpen}
  {title}
  class="{isOpen ? '' : 'hidden'} {clazz}"
  on:escPressed={() => (isOpen = false)}
  on:closeClicked={() => toggleOpen()}>
  <slot />
  {#if description}
    <div id="modal_desc">{description}</div>
  {/if}
</Dialog>
