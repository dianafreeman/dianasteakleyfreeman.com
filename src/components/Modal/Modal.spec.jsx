import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'
import Modal from '.';

describe('Modal', () => {

  it('should render without exploding', () => {
    expect(render(<Modal show/>)).not.to.throw
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Modal show/>).toJSON()
    jestExpect(tree).toMatchSnapshot();
  });
});
