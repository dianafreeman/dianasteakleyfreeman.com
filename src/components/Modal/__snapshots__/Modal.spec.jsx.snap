// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<div
  className="backdrop"
  style={
    Object {
      "backdropStyle": Object {
        "backgroundColor": "rgba(0,0,0,0.3)",
        "bottom": 0,
        "left": 0,
        "padding": 50,
        "position": "fixed",
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <div
    className="modal"
    style={
      Object {
        "modalStyle": Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "margin": "0 auto",
          "maxWidth": 500,
          "minHeight": 300,
          "padding": 30,
        },
      }
    }
  >
    <div
      className="footer"
    >
      <button
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
